plugins/inputs/t128_peer_path/t128_peer_path.go

```diff
@@ -94,25 +107,32 @@ func (plugin *T128PeerPath) Init() error {
 
 // Gather takes in an accumulator and adds the metrics that the Input gathers
 func (plugin *T128PeerPath) Gather(acc telegraf.Accumulator) error {
-	processedResponses := plugin.gqlCollector.MakeRequest(acc)
-	if processedResponses != nil {
-		for _, processedResponse := range processedResponses {
-			for k, v := range processedResponse.Tags {
-				if k == "adjacentAddress" && v == "127.117.97.105" {
-					delete(processedResponse.Tags, "adjacentAddress")
-					continue
-				}
-				if (k == "adjacentHostname" && v == "") || (plugin.ExcludeHostname) {
-					delete(processedResponse.Tags, "adjacentHostname")
-					continue
-				}
+	if !plugin.RetryIfNotFound && plugin.endpointNotFound {
+		return nil
+	}
+	processedResponses, err := plugin.gqlCollector.MakeRequest()
+	if err != nil && processedResponses == nil {
```

Comment:
`AddError` regardless of processResponses. If there's an error, it should be reported. That's the logic you've written anyway just with less clear dependencies. This is what you want.

```go
if err != nil {
    acc.AddError(err)
}
if processedResponses == nil {
    return nil
}
```

That being said, I believe `range` will do the correct thing if the slice is nil. So you should be able to simply:
```go
if err != nil {
    acc.AddError(err)
}

for _, processedResponse := range processedResponses {
    ...
}

return nil
```